script.sh

```diff
@@ -21,29 +25,28 @@ if [[ $INPUT_REEK_VERSION = "gemfile" ]]; then
   # if Gemfile.lock is here
   if [[ -f 'Gemfile.lock' ]]; then
     # grep for reek version
-    REEK_GEMFILE_VERSION=`cat Gemfile.lock | grep -oP '^\s{4}reek\s\(\K.*(?=\))'`
+    REEK_GEMFILE_VERSION=$(grep -oP '^\s{4}reek\s\(\K.*(?=\))' Gemfile.lock)
 
     # if reek version found, then pass it to the gem install
     # left it empty otherwise, so no version will be passed
     if [[ -n "$REEK_GEMFILE_VERSION" ]]; then
-      REEK_VERSION=$REEK_GEMFILE_VERSION
+      export REEK_VERSION=$REEK_GEMFILE_VERSION
       else
         printf "Cannot get the reek's version from Gemfile.lock. The latest version will be installed."
     fi
     else
       printf 'Gemfile.lock not found. The latest version will be installed.'
   fi
-  else
-    # set desired reek version
-    REEK_VERSION=$INPUT_REEK_VERSION
 fi
 
-gem install -N reek $(version $REEK_VERSION)
+# shellcheck disable=SC2046,SC2086
+sudo gem install --no-document reek $(version $REEK_VERSION)
```

Comment:
This change is breaking. Why using sudo here?

Reply:
Hi. Could you provide a bit more information how this change breaks your CI? sudo is used to system-wide install reek.

Reply:
It does not work on hosted runners. I think, using `ruby/setup-ruby` is overkill here, where the only need is to run linters.
`vterdunov/action-reek@do-not-use-sudo` worked fine :+1: